demos/classification_demo/grid_mat.hpp

```diff
@@ -0,0 +1,162 @@
+        cellSize = cv::Size(minCellSize, minCellSize);
+
+        for (int i = 0; i < size.height; i++) {
+            for (int j = 0; j < size.width; j++) {
+                points.emplace_back(cellSize.width * j, presenter.graphSize.height + cellSize.height * i);
+            }
+        }
+
+        outImg.create((cellSize.height * size.height) + presenter.graphSize.height,
+                       cellSize.width * size.width, CV_8UC3);
+        outImg.setTo(0);
+        
+        textSize = cv::getTextSize("", fontType, fontScale, thickness, &baseline);
+        accuracyMessageSize = cv::getTextSize("Accuracy (top 0): 0.000", fontType, fontScale, thickness, &baseline);
+        testMessageSize = cv::getTextSize(testMessage, fontType, fontScale, thickness, &baseline);
+    }
+
+    void textUpdate(double avgFPS, double avgLatency, double accuracy,
+                    bool isFpsTest, bool showAccuracy,
+                    Presenter& presenter) {
+        rectangle(outImg,
+                  {0, 0}, {outImg.cols, presenter.graphSize.height},
+                  cv::Scalar(0, 0, 0), cv::FILLED);
+
+        presenter.drawGraphs(outImg);
+        
+        cv::Scalar textColor = cv::Scalar(255, 255, 255);
+        int textPadding = 10;
+
+        cv::putText(outImg,
+                    cv::format("FPS: %0.01f", avgFPS),
+                    cv::Point(textPadding, textSize.height + textPadding),
+                    fontType, fontScale, textColor, thickness);
+        cv::putText(outImg,
+                    cv::format("Latency: %dms", static_cast<int>(avgLatency * 1000)),
+                    cv::Point(textPadding, (textSize.height + textPadding) * 2),
+                    fontType, fontScale, textColor, thickness);
+        
+        if (showAccuracy) {
+            cv::putText(outImg,
+                        cv::format("Accuracy (top %d): %.3f", FLAGS_nt, accuracy),
+                        cv::Point(outImg.cols - accuracyMessageSize.width - textPadding, textSize.height + textPadding),
+                        fontType, fontScale, textColor, thickness);
+        }
+
+        if (isFpsTest) {
+            cv::putText(outImg,
+                        testMessage,
+                        cv::Point(outImg.cols - testMessageSize.width - textPadding,
+                                  (textSize.height + textPadding) * 2),
+                        fontType, fontScale, cv::Scalar(50, 50, 255), thickness);
+        }
+    }
+
+    void updateMat(const std::list<LabeledImage>& imageInfos) {
+        size_t prevSourceId = (currSourceId + points.size() - prevImgs.size() % points.size()) % points.size();
+        while (!prevImgs.empty()) {
+            prevImgs.front().copyTo(outImg(cv::Rect(points[prevSourceId], cellSize)));
+            prevImgs.pop();
+            prevSourceId++;
+
+            if (prevSourceId >= points.size()) {
+                prevSourceId -= points.size();
+            }
+        }
```

Comment:
Could you explain the purpose of this loop? It looks like it redraws every image from the previous call again, but why?

Reply:
When I show images from the batch, I draw borders above them (cv::rectangle), but after the new batch of images arrives, I need to restore previous images (redraw them without borders). That's the purpose of this loop.

Reply:
OK, that makes sense. Could you add a comment to explain that, though?